core/zoe/contracts/coveredCallSrcs.js

```diff
@@ -0,0 +1,71 @@
+import harden from '@agoric/harden';
+
+const areAssaysValid = array => array.length === 2;
+
+const isValidInitialOfferDesc = newOfferDesc => {
+  const makeHasOkRules = validRules => offer =>
+    validRules.every((rule, i) => rule === offer[i].rule, true);
+
+  const hasOkLength = array => array.length === 2;
+  const hasOkRules = offer =>
+    makeHasOkRules(['offerExactly', 'wantExactly'])(offer) ||
+    makeHasOkRules(['wantExactly', 'offerExactly'])(offer);
+  return hasOkLength(newOfferDesc) && hasOkRules(newOfferDesc);
+};
+
+const makeWantedOfferDescs = firstOfferDesc => {
+  const makeSecondOffer = firstOffer =>
+    harden([
+      {
+        rule: firstOffer[1].rule,
+        assetDesc: firstOffer[0].assetDesc,
+      },
+      {
+        rule: firstOffer[0].rule,
+        assetDesc: firstOffer[1].assetDesc,
+      },
+    ]);
+  return harden([makeSecondOffer(firstOfferDesc)]);
+};
+
+const isValidOfferDesc = (extentOps, offerDescToBeMade, offerDescMade) => {
+  const ruleEqual = (leftRule, rightRule) => leftRule.rule === rightRule.rule;
```

Comment:
I keep getting tripped up by the part-vs-whole "rule" terminology. Is "rule" one of the strings `offerExactly`, etc...

Or is "rule" a record containing one of those strings and an assetDesc? 

I think you mostly mean the record, in which case the string might be "disposition"?